### ISSUE TITLE
valid-types false positive for type-guard syntax

### ISSUE THREAD

**Issue by Shinigami92** [gajus/eslint-plugin-jsdoc#567]
## Expected behavior

Should allow type guards

Working example:

![image](https://user-images.githubusercontent.com/7195563/83859298-00a6a480-a71e-11ea-83e2-3f71cc513100.png)

## Actual behavior

![image](https://user-images.githubusercontent.com/7195563/83859115-c6d59e00-a71d-11ea-80db-2066fad503cd.png)

## ESLint Config

```json
{
  "extends": ["plugin:jsdoc/recommended"]
}
```

## ESLint sample

```js
/**
 * String Type-Guard
 *
 * @param {any} value Value
 * @returns {value is string} `true` if value is of type string, otherwise `false`
 */
function isString(value) {
  return typeof value === "string";
}
```

## Environment

* Node version: 14.4.0
* ESLint version 7.1
* `eslint-plugin-jsdoc` version: 27.0.4

**Comment by brettz9** [comment]
I believe this is a dupe of https://github.com/jsdoctypeparser/jsdoctypeparser/issues/50 (our dependency for handling type parsing).
